source/plot/ShapePlotData2D.ooc

```diff
@@ -0,0 +1,92 @@
+use ooc-collections
+use ooc-math
+use ooc-draw
+import math
+import PlotData2D
+import svg/Shapes
+
+AnyShape: cover {
+	shape: Shape
+	region: FloatBox2D
+	colorBgra: ColorBgra
+	init: func@ (=shape, =region, =colorBgra)
+}
+
+ShapePlotData2D: class extends PlotData2D {
+	_shapes: VectorList<AnyShape>
+
+	init: func (label := "", colorBgra := ColorBgra new()) {
+		super(label, colorBgra)
+		this _shapes = VectorList<AnyShape> new()
+	}
+
+	addRectangle: func (box: FloatBox2D, colorBgra := ColorBgra new()) {
+		this _shapes add(AnyShape new(Shape Rectangle, box, colorBgra))
+	}
+
+	free: override func {
+		this _shapes free()
+		super()
+	}
+
+	getSvg: func (transform: FloatTransform2D) -> String {
+		result := ""
+		if (!this _shapes empty) {
+			noneColor := ColorBgra new(0, 0, 0, 0)
```

Comment:
@davidhesselbom will probably say something about this variable. Maybe `nullColor` or `emptyColor` ?